In [1]:
import pydataset

In [3]:
#요일별로 흡연자/비흡연자가 몇 명씩 방문하였는지 출력하세요.
tips = pydataset.data('tips')
tips

total_bill   tip     sex smoker   day    time  size
1         16.99  1.01  Female     No   Sun  Dinner     2
2         10.34  1.66    Male     No   Sun  Dinner     3
3         21.01  3.50    Male     No   Sun  Dinner     3
4         23.68  3.31    Male     No   Sun  Dinner     2
5         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
240       29.03  5.92    Male     No   Sat  Dinner     3
241       27.18  2.00  Female    Yes   Sat  Dinner     2
242       22.67  2.00    Male    Yes   Sat  Dinner     2
243       17.82  1.75    Male     No   Sat  Dinner     2
244       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [12]:
tips.groupby(['day','smoker']).count()

total_bill  tip  sex  time  size
day  smoker                                  
Fri  No               4    4    4     4     4
     Yes             15   15   15    15    15
Sat  No              45   45   45    45    45
     Yes             42   42   42    42    42
Sun  No              57   57   57    57    57
     Yes             19   19   19    19    19
Thur No              45   45   45    45    45
     Yes             17   17   17    17    17

In [16]:
#key1으로 묶고 각 그룹에서 data1의 평균을 구해보자
import pandas as pd
import numpy as np
df = pd.DataFrame({'key1': ['a', 'a', 'b',' b', 'a'],

                  'key2': ['one', 'two', 'one', 'two', 'one'],

                  'data1': np.random.randn(5),

                  'data2': np.random.randn(5)}) 
df

key1 key2     data1     data2
0    a  one -0.247833  0.241340
1    a  two  0.212572  0.021684
2    b  one -0.551107 -0.456421
3    b  two  1.140811  1.824525
4    a  one  0.373259 -1.343122

In [19]:
df.groupby('key1')[['data1']].mean()

data1
key1          
 b    1.140811
a     0.112666
b    -0.551107

In [21]:
#class별 manufacturer의 빈도수를 구하는 표를 만들어라.
mpg = pydataset.data('mpg')
mpg

manufacturer   model  displ  year  cyl       trans drv  cty  hwy fl  \
1           audi      a4    1.8  1999    4    auto(l5)   f   18   29  p   
2           audi      a4    1.8  1999    4  manual(m5)   f   21   29  p   
3           audi      a4    2.0  2008    4  manual(m6)   f   20   31  p   
4           audi      a4    2.0  2008    4    auto(av)   f   21   30  p   
5           audi      a4    2.8  1999    6    auto(l5)   f   16   26  p   
..           ...     ...    ...   ...  ...         ...  ..  ...  ... ..   
230   volkswagen  passat    2.0  2008    4    auto(s6)   f   19   28  p   
231   volkswagen  passat    2.0  2008    4  manual(m6)   f   21   29  p   
232   volkswagen  passat    2.8  1999    6    auto(l5)   f   16   26  p   
233   volkswagen  passat    2.8  1999    6  manual(m5)   f   18   26  p   
234   volkswagen  passat    3.6  2008    6    auto(s6)   f   17   26  p   

       class  
1    compact  
2    compact  
3    compact  
4    compact  
5    compact  
..       ...  
230  midsize  
231  midsize  
232  midsize  
233  midsize  
234  midsize  

[234 rows x 11 columns]

In [44]:
df = pd.crosstab(index=mpg['class'],columns=mpg['manufacturer'])
df

manufacturer  audi  chevrolet  dodge  ford  honda  hyundai  jeep  land rover  \
class                                                                          
2seater          0          5      0     0      0        0     0           0   
compact         15          0      0     0      0        0     0           0   
midsize          3          5      0     0      0        7     0           0   
minivan          0          0     11     0      0        0     0           0   
pickup           0          0     19     7      0        0     0           0   
subcompact       0          0      0     9      9        7     0           0   
suv              0          9      7     9      0        0     8           4   

manufacturer  lincoln  mercury  nissan  pontiac  subaru  toyota  volkswagen  
class                                                                        
2seater             0        0       0        0       0       0           0  
compact             0        0       2        0       4      12          14  
midsize             0        0       7        5       0       7           7  
minivan             0        0       0        0       0       0           0  
pickup              0        0       0        0       0       7           0  
subcompact          0        0       0        0       4       0           6  
suv                 3        4       4        0       6       8           0

In [49]:
df1 = mpg.reset_index()
df1

index manufacturer   model  displ  year  cyl       trans drv  cty  hwy  \
0        1         audi      a4    1.8  1999    4    auto(l5)   f   18   29   
1        2         audi      a4    1.8  1999    4  manual(m5)   f   21   29   
2        3         audi      a4    2.0  2008    4  manual(m6)   f   20   31   
3        4         audi      a4    2.0  2008    4    auto(av)   f   21   30   
4        5         audi      a4    2.8  1999    6    auto(l5)   f   16   26   
..     ...          ...     ...    ...   ...  ...         ...  ..  ...  ...   
229    230   volkswagen  passat    2.0  2008    4    auto(s6)   f   19   28   
230    231   volkswagen  passat    2.0  2008    4  manual(m6)   f   21   29   
231    232   volkswagen  passat    2.8  1999    6    auto(l5)   f   16   26   
232    233   volkswagen  passat    2.8  1999    6  manual(m5)   f   18   26   
233    234   volkswagen  passat    3.6  2008    6    auto(s6)   f   17   26   

    fl    class  
0    p  compact  
1    p  compact  
2    p  compact  
3    p  compact  
4    p  compact  
..  ..      ...  
229  p  midsize  
230  p  midsize  
231  p  midsize  
232  p  midsize  
233  p  midsize  

[234 rows x 12 columns]

In [57]:
df1.melt(id_vars = 'class', value_vars = df1.columns[2:]) 

class variable value
0     compact    model    a4
1     compact    model    a4
2     compact    model    a4
3     compact    model    a4
4     compact    model    a4
...       ...      ...   ...
2101  midsize       fl     p
2102  midsize       fl     p
2103  midsize       fl     p
2104  midsize       fl     p
2105  midsize       fl     p

[2106 rows x 3 columns]

```
value가 0~9는 few, 10~19는 many라는 새로운 열 frequency를 추가시켜라.
pd.cut 사용하기
```

In [66]:
a = df1.melt(id_vars = 'class', value_vars = df1.columns[2:])

bins = [0,10,20]
pd.cut(a['value'], bins=[0,10,20], right=False,
       labels=['few','many'])

TypeError: '<' not supported between instances of 'int' and 'str'

In [69]:
bins = [0,10,20]
label = ['few','many']
a['frequency'] = pd.cut(a['value'],bins = bins,
                          labels = label,right=False) 

TypeError: '<' not supported between instances of 'int' and 'str'

In [74]:
df2['frequency'] = pd.cut(x = df2['value'], bins = [0, 10, 20], 
                          labels = ['few', 'many'], right = False)
df2 

NameError: name 'df2' is not defined

In [71]:
import pydataset
mpg = pydataset.data('mpg')
mpg

#manufacturer과 year 별로 다른 모든 칼럼의 평균과 최댓값을 구해보세요.

manufacturer   model  displ  year  cyl       trans drv  cty  hwy fl  \
1           audi      a4    1.8  1999    4    auto(l5)   f   18   29  p   
2           audi      a4    1.8  1999    4  manual(m5)   f   21   29  p   
3           audi      a4    2.0  2008    4  manual(m6)   f   20   31  p   
4           audi      a4    2.0  2008    4    auto(av)   f   21   30  p   
5           audi      a4    2.8  1999    6    auto(l5)   f   16   26  p   
..           ...     ...    ...   ...  ...         ...  ..  ...  ... ..   
230   volkswagen  passat    2.0  2008    4    auto(s6)   f   19   28  p   
231   volkswagen  passat    2.0  2008    4  manual(m6)   f   21   29  p   
232   volkswagen  passat    2.8  1999    6    auto(l5)   f   16   26  p   
233   volkswagen  passat    2.8  1999    6  manual(m5)   f   18   26  p   
234   volkswagen  passat    3.6  2008    6    auto(s6)   f   17   26  p   

       class  
1    compact  
2    compact  
3    compact  
4    compact  
5    compact  
..       ...  
230  midsize  
231  midsize  
232  midsize  
233  midsize  
234  midsize  

[234 rows x 11 columns]

In [76]:
a = mpg.groupby(['manufacturer','year']).agg([np.mean,max])
a

C:\Users\admin\AppData\Local\Temp\ipykernel_11584\4228087442.py:1: FutureWarning: ['model', 'trans', 'drv', 'fl', 'class'] did not aggregate successfully. If any error is raised this will raise in a future version of pandas. Drop these columns/ops to avoid this warning.
  a = mpg.groupby(['manufacturer','year']).agg([np.mean,max])


displ            cyl            cty            hwy    
                       mean  max      mean max       mean max       mean max
manufacturer year                                                           
audi         1999  2.355556  2.8  5.111111   6  17.111111  21  26.111111  29
             2008  2.733333  4.2  5.333333   8  18.111111  21  26.777778  31
chevrolet    1999  4.971429  6.5  7.142857   8  15.142857  19  21.571429  27
             2008  5.116667  7.0  7.333333   8  14.916667  22  22.083333  30
dodge        1999  4.318750  5.9  6.750000   8  13.375000  18  18.437500  24
             2008  4.423810  5.7  7.333333   8  12.952381  17  17.571429  24
ford         1999  4.453333  5.4  7.066667   8  13.933333  18  18.600000  26
             2008  4.660000  5.4  7.400000   8  14.100000  17  20.500000  26
honda        1999  1.600000  1.6  4.000000   4  24.800000  28  31.600000  33
             2008  1.850000  2.0  4.000000   4  24.000000  26  33.750000  36
hyundai      1999  2.300000  2.5  4.666667   6  18.333333  19  26.666667  29
             2008  2.525000  3.3  5.000000   6  18.875000  21  27.000000  31
jeep         1999  4.350000  4.7  7.000000   8  14.500000  15  18.500000  20
             2008  4.650000  6.1  7.333333   8  13.166667  17  17.333333  22
land rover   1999  4.300000  4.6  8.000000   8  11.000000  11  15.000000  15
             2008  4.300000  4.4  8.000000   8  12.000000  12  18.000000  18
lincoln      1999  5.400000  5.4  8.000000   8  11.000000  11  16.500000  17
             2008  5.400000  5.4  8.000000   8  12.000000  12  18.000000  18
mercury      1999  4.500000  5.0  7.000000   8  13.500000  14  17.000000  17
             2008  4.300000  4.6  7.000000   8  13.000000  13  19.000000  19
nissan       1999  2.900000  3.3  5.333333   6  17.666667  21  23.500000  29
             2008  3.585714  5.6  5.714286   8  18.428571  23  25.571429  32
pontiac      1999  3.566667  3.8  6.000000   6  17.000000  18  26.333333  27
             2008  4.550000  5.3  7.000000   8  17.000000  18  26.500000  28
subaru       1999  2.400000  2.5  4.000000   4  19.000000  21  25.500000  26
             2008  2.500000  2.5  4.000000   4  19.500000  20  25.625000  27
toyota       1999  2.765000  4.7  5.000000   8  18.150000  26  24.050000  35
             2008  3.221429  5.7  5.285714   8  19.071429  28  26.142857  37
volkswagen   1999  2.206250  2.8  4.625000   6  21.250000  35  29.687500  44
             2008  2.327273  3.6  4.545455   6  20.454545  22  28.545455  29

In [83]:
#displ은 최댓값, cyl은 합을 구하기

.agg({'displ': max, 'cyl':sum})

KeyError: "Column(s) ['cyl', 'displ'] do not exist"

# 진도

In [91]:
df = mpg.iloc[:5, 2:5]
df

displ  year  cyl
1    1.8  1999    4
2    1.8  1999    4
3    2.0  2008    4
4    2.0  2008    4
5    2.8  1999    6

In [92]:
#1. df를 array로 바꾸기
df.values

array([[1.800e+00, 1.999e+03, 4.000e+00],
       [1.800e+00, 1.999e+03, 4.000e+00],
       [2.000e+00, 2.008e+03, 4.000e+00],
       [2.000e+00, 2.008e+03, 4.000e+00],
       [2.800e+00, 1.999e+03, 6.000e+00]])

In [96]:
#2. df를 딕셔너리로 바꾸기
df.to_dict()

{'displ': {1: 1.8, 2: 1.8, 3: 2.0, 4: 2.0, 5: 2.8},
 'year': {1: 1999, 2: 1999, 3: 2008, 4: 2008, 5: 1999},
 'cyl': {1: 4, 2: 4, 3: 4, 4: 4, 5: 6}}

In [101]:
#3. df를 리스트로 바꾸기
df.values.tolist() #리스트로 바꾸려면 데이터프레임일때는 X

[[1.8, 1999.0, 4.0],
 [1.8, 1999.0, 4.0],
 [2.0, 2008.0, 4.0],
 [2.0, 2008.0, 4.0],
 [2.8, 1999.0, 6.0]]

In [121]:
#df를 하나의 리스트로 만들기(가로로 줄 세우기)
li = []
for i in df.values.tolist():    #정석 방식
    for j in i:
        li.append(j)
li

[1.8,
 1999.0,
 4.0,
 1.8,
 1999.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.8,
 1999.0,
 6.0]

In [118]:
df.values.reshape(-1).tolist()   #같은방법

[1.8,
 1999.0,
 4.0,
 1.8,
 1999.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.8,
 1999.0,
 6.0]

In [120]:
df.values.flatten().tolist() #flatten() '평평하게 편다'

[1.8,
 1999.0,
 4.0,
 1.8,
 1999.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.0,
 2008.0,
 4.0,
 2.8,
 1999.0,
 6.0]

# 2. 머신러닝

In [2]:
import sklearn

In [3]:
sklearn.__version__

'1.0.2'

In [4]:
pip list | findstr sk

dask                          2022.2.1
Flask                         1.1.2
Note: you may need to restart the kernel to use updated packages.


In [10]:
from sklearn.datasets import load_iris
iris = load_iris()
iris

{'data': array([[5.1, 3.5, 1.4, 0.2],
        [4.9, 3. , 1.4, 0.2],
        [4.7, 3.2, 1.3, 0.2],
        [4.6, 3.1, 1.5, 0.2],
        [5. , 3.6, 1.4, 0.2],
        [5.4, 3.9, 1.7, 0.4],
        [4.6, 3.4, 1.4, 0.3],
        [5. , 3.4, 1.5, 0.2],
        [4.4, 2.9, 1.4, 0.2],
        [4.9, 3.1, 1.5, 0.1],
        [5.4, 3.7, 1.5, 0.2],
        [4.8, 3.4, 1.6, 0.2],
        [4.8, 3. , 1.4, 0.1],
        [4.3, 3. , 1.1, 0.1],
        [5.8, 4. , 1.2, 0.2],
        [5.7, 4.4, 1.5, 0.4],
        [5.4, 3.9, 1.3, 0.4],
        [5.1, 3.5, 1.4, 0.3],
        [5.7, 3.8, 1.7, 0.3],
        [5.1, 3.8, 1.5, 0.3],
        [5.4, 3.4, 1.7, 0.2],
        [5.1, 3.7, 1.5, 0.4],
        [4.6, 3.6, 1. , 0.2],
        [5.1, 3.3, 1.7, 0.5],
        [4.8, 3.4, 1.9, 0.2],
        [5. , 3. , 1.6, 0.2],
        [5. , 3.4, 1.6, 0.4],
        [5.2, 3.5, 1.5, 0.2],
        [5.2, 3.4, 1.4, 0.2],
        [4.7, 3.2, 1.6, 0.2],
        [4.8, 3.1, 1.6, 0.2],
        [5.4, 3.4, 1.5, 0.4],
        [5.2, 4.1, 1.5, 0.1],
  

In [15]:
from sklearn.datasets import load_iris

In [16]:
from sklearn.tree import DecisionTreeClassifier

In [17]:
from sklearn.model_selection import train_test_split

In [18]:
iris = load_iris()
iris

{'data': array([[5.1, 3.5, 1.4, 0.2],
        [4.9, 3. , 1.4, 0.2],
        [4.7, 3.2, 1.3, 0.2],
        [4.6, 3.1, 1.5, 0.2],
        [5. , 3.6, 1.4, 0.2],
        [5.4, 3.9, 1.7, 0.4],
        [4.6, 3.4, 1.4, 0.3],
        [5. , 3.4, 1.5, 0.2],
        [4.4, 2.9, 1.4, 0.2],
        [4.9, 3.1, 1.5, 0.1],
        [5.4, 3.7, 1.5, 0.2],
        [4.8, 3.4, 1.6, 0.2],
        [4.8, 3. , 1.4, 0.1],
        [4.3, 3. , 1.1, 0.1],
        [5.8, 4. , 1.2, 0.2],
        [5.7, 4.4, 1.5, 0.4],
        [5.4, 3.9, 1.3, 0.4],
        [5.1, 3.5, 1.4, 0.3],
        [5.7, 3.8, 1.7, 0.3],
        [5.1, 3.8, 1.5, 0.3],
        [5.4, 3.4, 1.7, 0.2],
        [5.1, 3.7, 1.5, 0.4],
        [4.6, 3.6, 1. , 0.2],
        [5.1, 3.3, 1.7, 0.5],
        [4.8, 3.4, 1.9, 0.2],
        [5. , 3. , 1.6, 0.2],
        [5. , 3.4, 1.6, 0.4],
        [5.2, 3.5, 1.5, 0.2],
        [5.2, 3.4, 1.4, 0.2],
        [4.7, 3.2, 1.6, 0.2],
        [4.8, 3.1, 1.6, 0.2],
        [5.4, 3.4, 1.5, 0.4],
        [5.2, 4.1, 1.5, 0.1],
  

In [20]:
type(iris) #딕셔너리와 같은것

sklearn.utils.Bunch

In [21]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

In [22]:
iris.values()

dict_values([array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
 

In [24]:
iris.items() #key와 values를 동시에 보여준다.

dict_items([('data', array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4,

In [26]:
d = {'apple':'사과', 'banana':'바나나'}
d

{'apple': '사과', 'banana': '바나나'}

In [28]:
d['apple']  #밑에와 같은 맥락

'사과'

In [29]:
iris_data = iris['data']
iris_data

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

In [33]:
iris_label = iris['target']
iris_label

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [38]:
X_train, X_test, y_train, y_test = train_test_split(iris_data, iris_label)

In [40]:
dt_clf = DecisionTreeClassifier()

In [63]:
dt_clf.fit(X_train, y_train)
#fitting 적합한다. 학습한다. 훈련한다. 

DecisionTreeClassifier()

In [43]:
X_train.shape

(112, 4)

데이터의 갯수는 112개라는 이야기.

칼럼.특질.피쳐는 4개이다.

In [45]:
y_train.shape

(112,)

In [46]:
dt_clf = DecisionTreeClassifier()

In [47]:
dt_clf.fit(X_train, y_train)

DecisionTreeClassifier()

In [52]:
X_test.shape  #38명을 다 맞춤

(38, 4)

In [55]:
y_pred = dt_clf.predict(X_test)
y_pred  #예측

array([0, 2, 0, 2, 2, 2, 0, 2, 1, 0, 1, 1, 0, 1, 0, 0, 2, 1, 2, 2, 1, 0,
       2, 1, 1, 1, 1, 1, 1, 0, 2, 0, 1, 2, 2, 1, 2, 0])

In [66]:
y_test  #정답

array([0, 2, 0, 2, 2, 2, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 2, 1, 2, 2, 1, 0,
       2, 2, 1, 1, 1, 1, 1, 0, 2, 0, 1, 2, 2, 1, 2, 0])

In [59]:
sum(y_pred == y_test)

36

In [60]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

In [62]:
iris.target_names    #위쪽에 0,1 이렇게 숫자로 써져있는건 이걸 나타냄.

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

In [64]:
from sklearn.metrics import accuracy_score  #정확도 점수

In [65]:
accuracy_score(y_test, y_pred)

0.9473684210526315

# LabelEncoder()

In [68]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()

In [71]:
labels = encoder.fit_transform(['TV', '냉장고', '전자레인지'])
labels

array([0, 1, 2], dtype=int64)